[Feature] add pool-aware file upload with default pool selection

Texas0295 opened the pull request:
1. Introduce general file upload support with a pool-aware tus client.
3. Add a pool fetching service and provider to retrieve available pools.
4. Extend settings and AppSettings to support default pool selection.
5. Add validation and fallback to exclude restricted pools and ensure a consistent default is always available.

With these changes, uploads now respect user-selected pools,
while maintaining backward compatibility by falling back to
the standard "Solar Network Driver" pool if no valid selection exists.

Texas0295 committed:
add general file upload support with pool-aware tus client

- add "uploadFile" i18n key (en, zh-CN, zh-TW)
- introduce putFileToPool for tus upload with X-FilePool header
- add ComposeLogic.pickGeneralFile for arbitrary files
- extend uploadAttachment to support poolId override
- add toolbar button for general file upload

Signed-off-by: Texas0295 <kimura@texas0295.top>

Texas0295 committed:
add pool fetching service and provider

- define FilePool model
- implement PoolService with /drive/pools endpoint
- add Riverpod providers (poolServiceProvider, poolsProvider)

Signed-off-by: Texas0295 <kimura@texas0295.top>

Texas0295 committed:
support default file pool selection

- add defaultPoolId to AppSettings + persistence
- extend SettingsScreen with pool dropdown
- update uploadAttachment to use defaultPoolId with fallback

Signed-off-by: Texas0295 <kimura@texas0295.top>

Texas0295 committed:
add default pool selection with validation and fallback

- extend AppSettings with defaultPoolId
- add pool filtering utility to exclude media-only pools
- add resolveDefaultPoolId with fallback to safe pool
- update SettingsScreen with default pool dropdown
- integrate uploadAttachment with default pool resolution

Signed-off-by: Texas0295 <kimura@texas0295.top>

LittleSheep2Code commented on lib/models/file_pool.dart:
The models should using freezed to build, and the remote types (from server) should have a prefix Sn, so it should be `SnFilePool`

LittleSheep2Code commented on lib/pods/pool_provider.dart:
```diff
@@ -0,0 +1,14 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import '../services/pool_service.dart';
```
Should use the package import, in dart, the package import and the file relative path import can produce two different type which cannot be converted. This might lead to some unexpected error in the future.

Texas0295 replied:
Refactored as suggested.

LittleSheep2Code commented on lib/services/file.dart:
```diff
@@ -40,6 +40,55 @@ Future<XFile?> cropImage(
   );
 }
 
+Completer<SnCloudFile?> putFileToPool({
```
Instead of creating a new method, use a optional argument might be a better way to handle this situation

Texas0295 replied:
Refactored accordingly, switched to optional argument.
Went a bit further to unify previous `putMediaToCloud` and `putFileToPool` under a single `putFileToCloud` with `FileUploadMode`.
This should cover both scenarios cleanly without breaking existing calls.

LittleSheep2Code commented on lib/services/pool_service.dart:
```diff
@@ -0,0 +1,19 @@
+
+import 'package:island/models/file_pool.dart';
+
+class PoolService {
```
Instead of building a service, building a riverpod provider might be a better way to provide pool data.

Texas0295 replied:
Refactored as suggested.
`PoolService` and `pool_utils` were removed, the fetching/filtering logic is now directly exposed through a `FutureProvider<List<SnFilePool>>` with an extension on `List<SnFilePool>` for `filterValid()`.

LittleSheep2Code commented on lib/utils/pool_utils.dart:
The utils method can be static members in the PoolService

Texas0295 replied:
Followed the same intention by moving `filterValidPools` into an extension method (`List<SnFilePool>.filterValid()`), so the logic is colocated with the model instead of a separate utils/service file.

LittleSheep2Code reviewed (changes requested):
The upstream (me) have updated to new file uploading protocol which aims to replace the tus. You can either continue editing this PR to fix some problems that I metioned or pull from upstream first.

The new upload protocol have a better way to handle the pool id when uploading.

Texas0295 committed:
migrate file upload from tus to FileUploader API

The tus-based upload flow (/drive/tus) has been removed upstream in favor
of a new multipart upload protocol. This commit replaces all TusClient
usage with the new FileUploader service that follows the official
/drive/files/upload/{create,chunk,complete} endpoints.

Changes include:
- remove tus_client_dart dependency and related code
- add putFileToPool() backed by FileUploader.uploadFile()
- update uploadAttachment() to call the new putFileToPool
- preserve poolId support, filename, and mimetype handling
- ensure progress callbacks fire at start and completion

This aligns the client with the new upload protocol while keeping the
same Compose UI and settings logic introduced in earlier patches.

Signed-off-by: Texas0295 <kimura@texas0295.top>

Texas0295 committed:
[REF] refactor file pool model and imports

- Refactor SnFilePool using freezed + sealed for consistency with other Sn models
- Add extension method listFromResponse for PoolService compatibility
- Update PoolService and utils to use SnFilePool
- Replace relative imports (../) with package imports for clarity and maintainability

Signed-off-by: Texas0295 <kimura@texas0295.top>

Texas0295 committed:
[REF] unify pool handling with extension methods

- Move pool filtering and parsing logic into SnFilePool extension
- Replace PoolService and pool_utils with unified extension
- Update settings screen to use pools.filterValid() + resolveDefaultPoolId
- Cleanup references in compose_shared.dart
- Remove obsolete files: pool_service.dart, pool_utils.dart

Signed-off-by: Texas0295 <kimura@texas0295.top>

Texas0295 committed:
[REF] unify file upload logic and pool utils

- merge putMediaToCloud and putFileToPool into putFileToCloud
  with FileUploadMode for media-safe vs generic uploads

Signed-off-by: Texas0295 <kimura@texas0295.top>

LittleSheep2Code commented:
LGTM and TYSM
